dask/optimization.py

```diff
@@ -567,7 +567,11 @@ def fuse(dsk, keys=None, dependencies=None, ave_width=None, max_width=None,
     reducible = {k for k, vals in rdeps.items() if len(vals) == 1}
     if keys:
         reducible -= keys
-    if not reducible:
+    if (not reducible and
+            (not fuse_subgraphs or not
+             {k for k, vals in rdeps.items() if len(set(vals)) == 1})):
```

Comment:
This might be slightly faster as `any(len(set(v)) == 1 for v in rdeps.values())` just to get the fast exit.  No idea if this is significant though.